### DISCUSSION TITLE
Fill gap time series

### DISCUSSION THREAD

**Discussion by spetton** [JoaquinAmatRodrigo/skforecast#284] [Q&A] [answered]
Hello,
I wonder if it is possible to fill gap of time series with the predict function. I have trained a MLP model to predict hourly sea water temperature with air temperature, day of year and several other exogenous variables and a lag window of 2 days. Sometimes in my data, I have missing data during several days. Is it possible to easily fill these gap automatically with last_window argument or do I need do create a loop over my dataset with manual prediction of the missing steps ?
Thanks for any help

**Comment by JoaquinAmatRodrigo** [comment, marked as answer]
Hi @spetton ,

Currently, skforecast models require the time series to be complete. You could try to train a forecaster with the longest portion of your time series without missing values, and then predict the missing gaps.

In the next release (0.6.0), all forecasters will allow the use of weights in the model fitting. Therefore, you can impute the missing values with a random value and give a weight of 0 to ignore them during the training. You can install the pre-release with  `pip install git+https://github.com/JoaquinAmatRodrigo/skforecast#master`.

Hope this helps!

**Reply by spetton** [reply]
Thank you for your quick reply ! I will try your solution

**Comment by JavierEscobarOrtiz** [comment]
Hi @spetton,

Another option is to fill in the gaps with a rolling average, train the model, predict the gaps, and then retrain the model with the filled data.

From my point of view, this might make sense if you have some exogenous variables that affect your forecaster.

Best,
